File: android/app/src/main/java/voidawt/event/WindowListener.java
package voidawt.event;

public interface WindowListener {
    void windowOpened(WindowEvent e);

    void windowClosing(WindowEvent e);

    void windowClosed(WindowEvent e);

    void windowIconified(WindowEvent e);

    void windowDeiconified(WindowEvent e);

    void windowActivated(WindowEvent e);

    void windowDeactivated(WindowEvent e);
}

File: src/jsMain/kotlin/awt/event/WindowListener.kt
package awt.event

actual interface WindowListener {
    actual fun windowOpened(e: WindowEvent?)
    actual fun windowClosing(e: WindowEvent?)
    actual fun windowClosed(e: WindowEvent?)
    actual fun windowIconified(e: WindowEvent?)
    actual fun windowDeiconified(e: WindowEvent?)
    actual fun windowActivated(e: WindowEvent?)
    actual fun windowDeactivated(e: WindowEvent?)
}
